provider/cldr/src/transform/time_zones/cldr_json/convert.rs

```diff
@@ -104,14 +116,8 @@ impl<'data> From<TimeZoneNames> for MetaZoneGenericNamesLongV1<'data> {
                         metazone
                             .long
                             .as_ref()
-                            .and_then(|long| {
-                                if long.0.len() == 1 {
-                                    long.0.values().next()
-                                } else {
-                                    long.0.get("generic")
-                                }
-                            })
-                            .map(|generic| (key.clone().into(), generic.clone().into()))
+                            .and_then(|zf| zf.0.get("generic").or_else(|| type_fallback(zf)))
```

Comment:
Suggestion (optional): `zf.0.get("generic")` is called here but also in `type_fallback(zf)`.  Consider directly calling `type_fallback(zf)`.

Reply:
My reasoning here is that this isn't particularly performance sensitive code, and this particular case is not going to be a big hit. 

I would rather have this code be more explicit in what it's trying to do than be more performant and cut corners. 

That is, I want the code to read, "I'm trying to get `generic` otherwise fallback" even if that has a redundancy in the fallback algorithm, rather than "fallback unconditionally without trying anything first because it just so happens that the fallback algorithm tries to get what we want first."

Reply:
Could you name the function something like `zf.get_generic_or_fallback()` ?

Reply:
I suppose I could, but getting the `generic` is really just part of the type fallback algorithm specified in UTS-35:
![image](https://user-images.githubusercontent.com/58569820/134051790-32067e77-33af-49f8-9812-3253ad8086f3.png)

It just so happens that in the DataProvider code, we don't have any cases where we "miss" on trying to get `daylight` or `standard` since we just collect whatever is available for those. 

I still don't see the redundancy here as a big issue. To me, it just makes it more clear that it's following the specification.